localstack-core/localstack/services/stepfunctions/asl/component/state/state_execution/state_task/service/state_task_service_api_gateway.py

```diff
@@ -174,6 +174,10 @@ def _headers_of(parameters: TaskParameters) -> Optional[dict]:
                 for forbidden_prefix in StateTaskServiceApiGateway._FORBIDDEN_HTTP_HEADERS_PREFIX:
                     if key.startswith(forbidden_prefix):
                         raise ValueError(f"The 'Headers' field contains unsupported values: {key}")
+
+                if isinstance(headers.get(key), list):
+                    headers[key] = str(headers.get(key))
```

Comment:
We could consider making the encoding slightly more deliberate here (comma separated string literals), perhaps using a json encoder or ','.join(<list>).